### DISCUSSION TITLE
User Login Issues post DB Migration (/identity/connect/token (login) => 400 Bad Request)

### DISCUSSION THREAD

**Discussion by mcollinsim** [dani-garcia/vaultwarden#1977] [Q&A]
We are attempting to migrate from Bitwarden to Vaultwarden, and are leaving the Bitwarden DB untouched on MySQL 5.7.33 so we have that for fallback.  After a bit of data massaging to account for the added fields in Vaultwarden, the data migration completes successfully.  Unfortunately, with the new Vaultwarden/DB, we are running into the user login issues (indicated with the log entry: /identity/connect/token (login) => 400 Bad Request).  I'm stumped, can anyone help?

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.22.2
* Web-vault version: v2.21.1
* Running within Docker: true
* Environment settings overridden: false
* Uses a reverse proxy: false
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 5.7.33-log
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": true,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "*****://******:********************************@*******-***-****-*.************.**-****-*.***.*********.***:****/*************",
  "db_connection_retries": 15,
  "disable_2fa_remember": true,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******.*****************.***",
  "domain_origin": "*****://******.*****************.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": false,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/arcanaaccesslog",
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": "61443",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

**Comment by mcollinsim** [comment]
To update, I've verified the time settings as well.

**Comment by bokkabonga** [comment]
Are you talking about Bitwarden or Bitwarden_rs (which is now vaultwarden)? And you didn´t change the Database type from SQLite to MySQL?

**Reply by mcollinsim** [reply]
Thanks for checking on that.  It's Bitwarden_rs, and we're moving to Vaultwarden.  We are running both on the same Mysql database/host, trying for separate schemas.

**Reply by mcollinsim** [reply]
Anyone else have a MYSQL to MYSQL DB migration issue for Vaultwarden?

**Comment by BlackDex** [comment]
What i think you did is the following.
You had an old, still named Bitwarden_rs, server running.
You wanted to go to the new server version and thus created a whole new setup including database.
After that you tried to copy over the old database to the new version and because the database schema is different you manually tried to fix it.

If that is the case, in short, that will probably not work.
There probably where some data migrations done which you didn't do.

Best thing you can do is the following.
1. Delete and purge the whole new setup 
2. Create a full copy of your current database so that you have two times the same data, which i think you wanted because of safety.
3. Now create the Vaultwarden instance and point that to this new copied over database.
4. During startup of Vaultwarden it will defect that it is an older versions and it will do all the data migrations for you.
